### ISSUE TITLE
MultilineMethodCallBraceLayout with new_line style false positive

### ISSUE THREAD

**Issue by maxjacobson** [bbatsov/rubocop#3462]
I believe the following is a false positive offense in this cop when it's configured this way:

## Expected behavior

My .rubocop.yml:

```yml
MultilineMethodCallBraceLayout:
  EnforcedStyle: new_line
```

My ruby:

```ruby
feed(
  'dog food', [
    dog1,
    dog2
  ].join(' ')
)
```

Expect no offenses

## Actual behavior

```shell
$ rubocop -D
Inspecting 1 file
C

Offenses:

indent.rb:5:13: C: Style/MultilineMethodCallBraceLayout: Closing method call brace must be on the line after the last argument.
  ].join(' ')
            ^

1 file inspected, 1 offense detected
```

## Workaround

I can satisfy the cop by changing my ruby to this:

```ruby
feed(
  'dog food', [
    dog1,
    dog2
  ].join(
    ' '
  )
)
```

But I don't think it should be necessary, because the `join` method call is not multiline
## Steps to reproduce the problem

Using the above yaml and ruby code and running rubocop should reproduce the same behavior

## RuboCop version

Include the output of `rubocop -V`:

```
$ rubocop -V
0.42.0 (using Parser 2.3.1.2, running on ruby 2.3.1 x86_64-darwin14)
```
